Sources/MapboxMaps/Annotations/AnnotationOrchestrator.swift

```diff
@@ -132,12 +147,23 @@ public class AnnotationOrchestrator {
             fatalError("DisplayLinkCoordinator must be present when creating an annotation manager")
         }
 
-        return CircleAnnotationManager(id: id,
+        let annotationManager = CircleAnnotationManager(id: id,
                                        style: style,
                                        singleTapGestureRecognizer: singleTapGestureRecognizer,
                                        mapFeatureQueryable: mapFeatureQueryable,
                                        shouldPersist: true,
                                        layerPosition: layerPosition,
                                        displayLinkCoordinator: displayLinkCoordinator)
+
+        annotationManagersById[id] = annotationManager
+
+        return annotationManager
+    }
+
+    /// Removes an annotation manager, this will remove the underlying layer and source from the style.
+    /// A removed annotation manager will not be able to reuse anymore, you will need to create new annotation manger to add annotations.
+    /// - Parameter id: Identifer of annotation manager to remove
+    public func removeAnnotationManager(withId id: String) {
+        annotationManagersById[id] = nil
```

Comment:
Rather than set it to nil, can we just remove it all together? using the dictionary [remove function](https://developer.apple.com/documentation/swift/dictionary/2831050-remove)